### PR TITLE
Added multithreading

### PR DESCRIPTION
**Pull request by crystalhsieh**
Added multithreading functionality. Fixed testing too.
Might add a handler to test multithreading as working

TODO: Add the number of threads we want to have - right now it caps at 5.

### CODE REVIEW

File: src/server.cc [hunk 1]
```diff
@@ -59,6 +59,24 @@ void Server::start_accept() {
           boost::asio::placeholders::error));
 }
 
+// Run the server in threads
+void Server::run() {
+  // Modeled after team AAAAA from class, which was shown in class code reviews
+  
+  // List of unique pointers to each thread
+  std::vector<boost::shared_ptr<boost::thread>> list_of_threads;
```
Comment (ryanvoong) [inline src/server.cc]: your comment says unique pointers but these are shared pointers. shouldn't they be unique pointers?

File: src/server.cc [hunk 2]
```diff
@@ -59,6 +59,24 @@ void Server::start_accept() {
...
+  for (std::size_t i = 0; i < NUM_OF_THREADS_; i++) {
+    boost::shared_ptr<boost::thread> thread(new boost::thread(
+      boost::bind(&boost::asio::io_service::run, &acceptor_.get_io_service())));
```
Comment (ryanvoong) [inline src/server.cc]: indent 4 spaces instead of 2

**Review by ryanvoong — APPROVED** [review]
looks good overall

File: src/server.cc (hunk 1 above)
Comment (crystalhsieh) [reply src/server.cc]: std::unique pointers aren't compatible with the threading from boost, for some reason..

**Pull request merged**